examples/jsm/nodes/materials/MeshStandardNodeMaterial.d.ts

```diff
@@ -1,7 +1,21 @@
+import { Node } from '../core/Node';
+import { ColorNode } from '../inputs/ColorNode';
+import { FloatNode } from '../inputs/FloatNode';
+import { Vector2Node } from '../inputs/Vector2Node';
 import { NodeMaterial } from './NodeMaterial';
 
 export class MeshStandardNodeMaterial extends NodeMaterial {
 
 	constructor();
 
+  color: ColorNode | Node;
```

Comment:
Does something like this really make sense? I mean an instance of `ColorNode` is also of type `Node`. So why not just using `Node`?

Reply:
You are right, but it gives more information about the equivalent type in shaders (`f`, `vec3`...)

Reply:
What do you think?

Reply:
Well, I don't know^^. I'm not using TS. The only thing I can say is that this type definition "looks" strange. I also think that not all classes derived of node make sense in context of `color`. That's probably true for most other properties.

So instead of using `Node`, it makes probably more sense to enumerate all node types which are semantically correct. E.g.
```ts
color: ColorNode | FunctionCallNode | OtherNode; // ...
```

Reply:
That's more work to put in place, but I totally agree with this.

Maybe we should define some types in order to factorize this code. Something like:
```typescript
type ColorInputNode: ColorNode | FunctionCallNode | Vector3Node ...;
```

This way we can reuse `ColorInputNode` everywhere it's needed

Reply:
Yeah, it definitely makes sense to define this stuff once and then use it in the code base.

Reply:
> So instead of using Node, it makes probably more sense to enumerate all node types which are semantically correct. E.g.

The difference in this case is that `NodeMaterial` convert any `Node` to `ColorNode` case you dont't use `ColorNode` this is equivalent to `parameter: string | object`, `parameter: DefaultNode | NodeBased`, similar what happens in `method overloading` in c# for example,  
```C#
void setInputX( value: ColorNode ) { ... }
void setInputX( value: Node ) { 
this.setInputX( ConvertToColorNode( value ) ) 
}
```


  Assume that we must declare all classes possible here, is considered almost every class in the NodeMaterial lib, even so fail in considered from third-parties Node extended.

Reply:
> so fail in considered from third-parties Node extended

True, I have myself some custom Node classes that I might contribute upstream, but if we don't set the attribute type to be something **OR** `Node` I won't be able to use it anymore.

Reply:
> I also think that not all classes derived of node make sense in context of color. That's probably true for most other properties.

@sunag so what do you think we should do with this issue? It seems to me that it will be difficult to validate at transpilation that a given `Node` is a valid `color` input. But I might be wrong. For example, a `FunctionCallNode` can be a valid color if it returns a `vec3`, but it won't be the case if the defined function returns a `float`... 
And we can't seem to know that at transpilation. Except if make the output type as part of the `Node` type, like `Node<ShaderType>` where `FunctionCallNode<Vec3>` would be a valid color but not `FunctionCallNode<Float>`.

Reply:
I would that @Mugen87 decide about it... even because he is the biggest responsible for the implementation of TypeScript in three.js. The absence of documentation of NodeMaterial difficult 
 the collective discrimination but we are walking for this improvement... I know that `Node` class in type is necessary for now, maybe could use multiples `interface` to flags the inputs and outputs? I still not thought in anything better for this.... Anyware your first approach typing only the Node doesn't describe well the type of input is 100% compatible `color: Node;`

Reply:
I think I would just use `Node` as a type and avoid the usage of more concrete classes for now. It feels technically more correct to specify just `color: Node` than `Node: ColorNode | Node`.

Reply:
@martinRenou We can follow the approach and use primitives values in `MeshStandardNodeMaterial` ( https://github.com/mrdoob/three.js/pull/18061#discussion_r355086139 )? Whay do you think? Sorry the extra work.

Reply:
> It feels technically more correct to specify just color: Node than color: ColorNode | Node.

Definitely. We can think of a better input validation at transpilation time later.

No worries @sunag, I'll do it :)